youtube_dl/extractor/common.py

```diff
@@ -1014,6 +1014,12 @@ def _extract_m3u8_formats(self, m3u8_url, video_id, ext=None,
             return []
         m3u8_doc, urlh = res
         m3u8_url = urlh.geturl()
+        is_media_playlist = re.search(r'#EXT-X-TARGETDURATION', m3u8_doc)
```

Comment:
This heuristic should be clarified with rationale comment. It can also be simplified to: 
```python
if '#EXT-X-TARGETDURATION' in m3u8_doc:
```
It's also probably makes sense to additionaly check for missing `#EXT-X-STREAM-INF`.